src/core_plugins/vega/public/vega_view/vega_base_view.js

```diff
@@ -329,6 +330,43 @@ export class VegaBaseView {
     return { from, to, mode };
   }
 
+  /**
+   * Set global debug variable to simplify vega debugging in console. Show info message first time
+   */
+  setDebugValues(view, spec, vlspec) {
+    if (!window) {
+      return;
+    }
+
+    if (window.VEGA_DEBUG === undefined && console) {
+      console.log('%cWelcome to Kibana Vega Plugin!', 'font-size: 16px; font-weight: bold;');
+      console.log('You can enable debugging by putting `%debug%: chooseAnyIdHere` into the top level of our spec.');
+      console.log('You can then access the Vega view for this chart with VEGA_DEBUG.chooseAnyIdHere' +
+        'Learn more at https://vega.github.io/vega/docs/api/debugging/.');
+      window.VEGA_DEBUG = {
+        VEGA_VERSION: vega.version,
+        VEGA_LITE_VERSION: vegaLite.version,
+      };
```

Comment:
what if we just add the `VEGA_DEBUG` window object only if user added the `%debug%` into the specs?
To avoid pollute the window with an object that is not required if we are not debugging.

A second, minor, useless 😜 thought is: what about using a different debug object name? maybe more "kibana" specific like `KIBANA_VEGA_DEBUG` or similar. So in the case, someone add a new plugin with a vega visualization you can avoid collision? maybe this is more a question for @nyurik.

Reply:
Could do this. My thought was since the version and lite version are sitting on that object we could already attach it at this point, but I am totally unopinionated about that, so can remove it if you want.

Regarding the renaming: I wanted to keep that in sync with the existing API, so we don't need to change the global name for that, which might make discuss questions more complicated. "Could you please copy `window.KIBANA_VEGA_DEBUG`" - "That doesn't exist" - "Oh which version are you running?" - "6.3.0" - "Okay, in this case `window.VEGA_DEBUG`.". But I am open for both, so let's @nyurik decide on this one :)

Reply:
I am not sure adding a new, temporary `%debug%` makes much sense for the following reasons:
* Harder to debug, just like Tim said, it would be "it doesn't work because you now have to  set %debug% to "xxx", and use `VEGA_DEBUG.xxx` instead of `VEGA_DEBUG`
* Doesn't fully solve the problem -- if user adds multiple instances of the same graph to the dashboard, they still conflict.
* renaming `VEGA_DEBUG` would be even worse - per Tim's comment.

On the other hand, it might be good to have %debug% as an advanced optional parameter - e.g. if set, use it as a subfield, and if not, use it, but again, I think we are solving a problem that doesn't need to be solved. Let's adhere to KISS, simply override the VEGA_DEBUG value as we did before, even if there are multiple graphs, and just implement the proper debug panel view.  Simpler in this case is much better than thorough but confusing to users.